ConnectionTask: Fix request failure retrying

AndrejMitrovic opened the pull request:
This might resolve the frequent CircleCI failures we've seen recently.

Fixes #1105

AndrejMitrovic committed:
ConnectionTask: Fix request failure retrying

Previously if the request failed after X number of
attempts this would throw an Exception which would
not be caught in the ConnectionTask.

Instead the request should be continuously retried
until the request was successful or until the
BanManager bans the node.

Fixes #1105

Geod24 commented on source/agora/network/NetworkManager.d:
```diff
@@ -162,11 +162,14 @@ public class NetworkManager
-                    // 404 => API not implemented, it's a FullNode
-                    if (ex.status == 404)
-                        break;
+                    if (auto http = cast(HTTPStatusException)ex)
+                    {
+                        // 404 => API not implemented, it's a FullNode
+                        if (http.status == 404)
+                            break;
+                    }
```
How does it work with LocalRest ?

AndrejMitrovic replied:
It's the NetworkClient which throws the exception if there was a timeout.

Without this change and with the test-case I've added ordinarily this would happen:

```
#### FATAL ERROR: Request failure to Validator #2 (GDC2...YD36) after 2 attempts
This node was started at source/agora/test/Base.d:459
This most likely means that the node crashed due to an uncaught exception
If not, please file a bug at https://github.com/Geod24/localrest/
```

AndrejMitrovic committed:
Decrease per-request timeout and increase max retries

The per-request timeout is too high. The waitForDiscovery()
call in the integration test only waited 5 seconds,
but this doesn't give us enough time to attempt a request again.

AndrejMitrovic commented:
The CI still failed so I'm trying another thing in a second commit.

AndrejMitrovic commented:
> The CI failed

But the "uncaught exception" errors are now gone so this is good.
